tests/fix/ProposeFunction.t/run.t

```diff
@@ -0,0 +1,26 @@
+  $ dune build
+  $ zanuda -dir . > /dev/null 
+  $ cat fix_gen/diffs.log
```

Comment:
As mentioned in the original issue, my desired workflow would be 

```
  $ dune build @lint # https://dune.readthedocs.io/en/latest/reference/aliases.html#lint
  $ dune promote
```

Dune already supports promoting diffs generated by lint tools as shown by this test

https://github.com/ocaml/dune/blob/9c41108359ac93d369a6e78da88ffbde8b1f8ca9/test/blackbox-tests/test-cases/lint.t/run.t#L12

I think making the first `dune build @lint` command work will require a design discussion about the best integration with dune. 

I think zanuda can make an incremental step forward towards that integration if it supports the following workflow

```suggestion
  $ dune build
  $ zanuda -dir . > /dev/null 
  $ dune promote
```

This will require writing files with the suffix ".lint-corrected" next to the original source files in the dune build directory in a promotable format that dune understands. The advantages are - cross platform support without having to generate a shell script.

Reply:
Thanks for your attention. Unfortunately, as discussed in https://github.com/ocaml/dune/issues/3471, dune doesn't support promotion for custom lints. 

> However, it seems that the definition of the promotion is only done in the Pps branch of the lint_module pattern_matching. 

The implementation of this feature was discussed below in the comments, but it seems that the goal was not achieved, since I didn't see the associated PRs and it was also not possible to achieve the desired behavior locally.
In your [test example](https://github.com/ocaml/dune/blob/9c41108359ac93d369a6e78da88ffbde8b1f8ca9/test/blackbox-tests/test-cases/lint.t/correct/dune), lint was also used in combination with pps, which gave the expected behavior:
`(lint (pps correct_static_add))`
That is, at this stage, without dune support, this seems to be almost the only solution, but I hope that in the future this situation will change.

If I still missed something, I'll be glad to your advice :)

Reply:
To defend from github email notification issues, it should be safe to highlight @petr-tik explicilty